api/base/views.py

```diff
@@ -75,3 +79,16 @@ def root(request, format=None):
             'users': absolute_reverse('users:user-list'),
         }
     })
+
+
+class OsfAPIViewMeta(type):
+    """Metaclass to ensure a base level of permissions are applied to all of our views
+    """
+    def __new__(cls, name, bases, attributes):
+        permission_classes = set((
+            drf_permissions.IsAuthenticatedOrReadOnly,
+            OsfBasePermission
```

Comment:
I think it would also make sense to add `base_permissions.TokenHasScope` and remove it from the view methods.